src/server/hset_family.cc

```diff
@@ -443,10 +484,16 @@ OpResult<vector<OptStr>> OpHMGet(const OpArgs& op_args, std::string_view key, Cm
     StringMap* sm = GetStringMap(pv, op_args.db_cntx);
 
     for (size_t i = 0; i < fields.size(); ++i) {
-      auto it = sm->Find(ToSV(fields[i]));
-      if (it != sm->end()) {
+      if (auto it = FindWithCleanup(
+              {.sm = sm, .db_slice = db_slice, .op_args = op_args, .key = key}, ToSV(fields[i]));
+          it != sm->end()) {
         result[i].emplace(it->second, sdslen(it->second));
       }
+
+      // If the set is empty, FindWithCleanup will have removed the key
+      if (sm->Empty()) {
```

Comment:
If FindWithCleanup removed the key, is sm still valid or was it deallocated?

Reply:
Good point, oddly enough in the debugger the pointer is still valid although its size is 0 and `entries_` table is empty although the block of memory seems to have been freed. It doesn't seem safe though.

I will look into this further.

Reply:
In the latest version of the code, the map is not accessed after delete because delete is performed during scope exit of the function.